src/Extensions/hcox_volcano_mod.F90

```diff
@@ -563,18 +565,20 @@ SUBROUTINE ReadVolcTable( HcoState, ExtState, Inst, RC )
     ! ReadVolcTable begins here!
     !=================================================================
 
-    ! Do only if it's a new day...
-    IF ( HcoClock_NewDay( HcoState%Clock, EmisTime=.TRUE. ) ) THEN
-
-       ! Get current year, month, day
-       CALL HcoClock_Get ( HcoState%Clock, cYYYY=YYYY, cMM=MM, cDD=DD, RC=RC )
-       IF ( RC /= HCO_SUCCESS ) RETURN
-
+    ! Get current year, month, day
+    CALL HcoClock_Get ( HcoState%Clock, cYYYY=YYYY, cMM=MM, cDD=DD, RC=RC )
```

Comment:
Could you give a brief explanation on why this change is here?

Reply:
This change was motivated by the same issue of calling `SetHcoTime` twice. This had the effect that `HcoClock_NewDay( HcoState%Clock, EmisTime=.TRUE. )` would always return .FALSE. and the volcano emissions would thus never be updated. I changed the code so that it now keeps track of the emissions data day that is in memory (variable `YmdOnFile`) and then compares the current simulation day against that. This is much cleaner, in my opinion, and works for all situations (first time step, rewinding the clock, etc.).